docs/sphinx/examples/cpp/other/builder/builder.cpp

```diff
@@ -114,6 +122,37 @@ int main() {
     counts.dump();
   }
 
+  {
+    // In a simulated environment, it is sometimes useful to be able to
+    // specify an initial state vector. The initial state vector is 2 to the
+    // power `n` where `n` is the number of qubits.
+
+    // In this example, we create a kernel template `sim_kernel` that captures
+    // the variable `init_state` by reference.
+    auto sim_builder = cudaq::make_kernel();
+    std::vector<cudaq::simulation_scalar> init_state;
+    auto q = sim_builder.qalloc(init_state);
```

Comment:
nit: I think it would be worth spelling out this variable name to implicitly remind people that the return type here is still qubit/s.

Reply:
This is following the convention used in the tests in this file. I agree that it could be improved. I'll file a PR.